### ISSUE TITLE
Shorten HTML anchor info text and simplify explanation

### ISSUE THREAD

**Issue by hanneslsm** [WordPress/gutenberg#48062]
## What problem does this address?
The info text for the HTML Anchor seems a bit long at the moment. 
It's taking up quite a bit of space (5 lines of text, plus margin-top of `a`, and 1 line for the link), which also affect the visual hierarchy. 
Plus, do we really need to explain what an HTML anchor is when there's a "learn more" option available?

<img width="279" alt="Screenshot 2023-02-14 at 07 55 30" src="https://user-images.githubusercontent.com/17861443/218661830-6d20853e-7f52-4bfa-b283-ff48578956db.png">

## What is your proposed solution?
Max 3 lines of text. 
<img width="141" alt="image" src="https://user-images.githubusercontent.com/17861443/218662925-84b727af-b857-4b94-85b7-1f15762101ec.png">  <img width="270" alt="image" src="https://user-images.githubusercontent.com/17861443/218662355-8a121d75-3f3f-4411-a4fd-ec585c665509.png">

**Comment by ghost** [comment]
I agree that it could be shorter, but I don't think the proposed text is what it should be. "Create a word" and "web address" don't sound right to me. Also, just having "Learn more" as a link is not in line with accessibility best practices.

I suggest changing the text to:
"Enter a unique word or phrase (without spaces), so you can link directly to this section of the page. _Learn more about anchors._"
